packages/jaeger-ui/src/components/DeepDependencies/Header/LayoutSettings/index.test.js

```diff
@@ -48,6 +48,26 @@ describe('LayoutSettings', () => {
     jest.clearAllMocks();
   });
 
+  it('uses value found in localStorage', () => {
+    const selectedDensity = props.density;
+    localStorage.setItem(LayoutSettings.DEFAULT_DENSITY, selectedDensity);
+    const radios = getWrapper().find(Radio);
+    expect(Array.from(radios).findIndex(radio => radio.props.checked)).toBe(densityIdx);
+  });
+
+  it('records new selection in localStorage', () => {
+    const newIdx = 2;
+    const newDensity = densityOptions[newIdx].option;
+    getWrapper()
+      .find(Radio)
+      .at(newIdx)
+      .simulate('change', { target: { value: newDensity } });
+    expect(props.setDensity).toHaveBeenCalledWith(newDensity);
+    expect(trackDensityChangeSpy).toHaveBeenCalledWith(props.density, newDensity, densityOptions);
+    // Ensure the value is recorded in localStorage
+    expect(localStorage.getItem(LayoutSettings.DEFAULT_DENSITY)).toBe(newDensity);
```

Comment:
I would recommend setting local storage slot to null at the start of the test, this way you can be sure that it is actually changing and wasn't already holding your desired value. It would make the test self contained.